onnx/defs/__init__.py

```diff
@@ -47,6 +62,55 @@ def _attribute_default_value(self):  # type: ignore
 OpSchema.Attribute.default_value = _attribute_default_value  # type: ignore
 
 
+def _op_schema_repr(self) -> str:
+    return f"""\
+OpSchema(
+    name={self.name!r},
+    domain={self.domain!r},
+    since_version={self.since_version!r},
+    doc={self.doc!r},
+    type_constraints={self.type_constraints!r},
+    inputs={self.inputs!r},
+    outputs={self.outputs!r},
+    attributes={self.attributes!r}
+)"""
+
+
+OpSchema.__repr__ = _op_schema_repr  # type: ignore
+
+
+def _op_schema_formal_parameter_repr(self) -> str:
```

Comment:
Why not implementing them in C++?

Reply:
I think it is easier and cleaner in Python. For example, when repr() is called on OpSchema, it is automatically called on fields that contain data structures like FormalParameter etc. If we do it in c++, we need to hand format lists and call repr on nested data structure etc. I think? 

Also if anyone wants to change the code to prototype, they wouldn’t need to recompile from source.

Reply:
Marking as resolved. LMK if you have further comments, thanks!